### PR TITLE
clean up formatting  in cosmology docs

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

Just some cleaning. Minimal wording change. Mostly getting stuff to same wrap length.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/cosmology/dev.rst [hunk 1]
```diff
@@ -6,24 +6,22 @@ Cosmology For Developers
 Cosmologies in Functions
 ========================
 
-It is often useful to assume a default cosmology so that the exact
-cosmology does not have to be specified every time a function or method
-is called. In this case, it is possible to specify a "default"
-cosmology.
+It is often useful to assume a default cosmology so that the exact cosmology
+does not have to be specified every time a function or method is called. In
```
Comment (mhvk) [inline docs/cosmology/dev.rst:10]: I'd delete the last sentence (In this case, ...) - superfluous given what follows.

File: docs/cosmology/dev.rst [hunk 2]
```diff
@@ -6,24 +6,22 @@ Cosmology For Developers
...
+|default_cosmology| to set a cosmology for the current Python session. If you
+have not set a default cosmology using one of the methods described above, then
+the cosmology module will default to using the
 ``default_cosmology._value`` parameters.
 
-It is strongly recommended that you use the default cosmology through
-the :class:`~astropy.cosmology.default_cosmology` science state object. An
-override option can then be provided using something like the
-following:
+It is strongly recommended that you use the default cosmology through the
```
Comment (mhvk) [inline docs/cosmology/dev.rst:22]: Why not just "You can override the default using ..."?

File: docs/cosmology/dev.rst [hunk 3]
```diff
@@ -35,16 +33,15 @@ following:
...
         ... function code here ...
 
-This ensures that all code consistently uses the default cosmology
-unless explicitly overridden.
+This ensures that all code consistently uses the default cosmology unless
+explicitly overridden.
 
 .. note::
 
     In general it is better to use an explicit cosmology (for example
```
Comment (mhvk) [inline docs/cosmology/dev.rst:41]: This seems very strange after the above - the user has just been told to use a default cosmology and now that is reversed! Maybe "If you are preparing a paper and thus need to ensure your code provides reproducible results, it is better to ..."

File: docs/cosmology/dev.rst [hunk 4]
```diff
@@ -35,16 +33,15 @@ following:
...
-    ``WMAP9.H(0)`` instead of
-    ``cosmology.default_cosmology.get().H(0)``). Use of the default
-    cosmology should generally be reserved for code that will be
-    included in the ``astropy`` core or an affiliated package.
+    ``WMAP9.H(0)`` instead of ``cosmology.default_cosmology.get().H(0)``).
+    Use of the default cosmology should generally be reserved for code that
```
Comment (mhvk) [inline docs/cosmology/dev.rst:43]: Why? Either remove or explain better

File: docs/cosmology/dev.rst [hunk 5]
```diff
@@ -174,29 +168,27 @@ Speeding up Integrals in Custom Cosmologies
...
+key thing is to explicitly set the instance variables
 ``self._inv_efunc_scalar`` and ``self._inv_efunc_scalar_args`` in the
 constructor for the subclass, where the latter are all the arguments except
 ``z`` to ``_inv_efunc_scalar``. The provided classes do use this optimization,
 and in fact go even further and provide optimizations for no radiation, and for
-radiation with massless neutrinos coded in cython. Consult the
-:class:`~astropy.cosmology.FLRW` subclasses for details, and
-``scalar_inv_efuncs`` for the details.
+radiation with massless neutrinos coded in cython. Consult the |FLRW|
+subclasses for details, and ``scalar_inv_efuncs`` for the details.
```
Comment (mhvk) [inline docs/cosmology/dev.rst:191]: remove one of the "for details"

File: docs/cosmology/dev.rst [hunk 6]
```diff
@@ -205,18 +197,16 @@ Astropy Interoperability: I/O and your Cosmology Package
 ========================================================
 
 If you are developing a package and want to be able to interoperate with
-`astropy.cosmology.Cosmology`, you're in the right place! Here we will discuss
-and provide examples for enabling Astropy to read and write your file formats,
-but also convert your cosmology objects to and from Astropy's |Cosmology|.
+|Cosmology|, you're in the right place! Here we will discuss and provide
+examples for enabling Astropy to read and write your file formats, but also
```
Comment (mhvk) [inline docs/cosmology/dev.rst:201]: I'd say "discuss how to enable ... formats, and convert ..."

File: docs/cosmology/index.rst [hunk 7]
```diff
@@ -51,18 +60,16 @@ transverse proper kiloparsecs (kpc) corresponding to an arcminute at z=3::
...
-IPython you can also use tab completion to print a list of the
-available methods. To do this, after importing the cosmology as in the
-above example, type ``cosmo.`` at the IPython prompt and then press
-the tab key.
+Here |WMAP9| is a built-in object describing a cosmology with the parameters
+from the nine-year WMAP results. Several other built-in cosmologies are also
+available (see `Built-in Cosmologies`_). The available methods of the cosmology
+object are listed in the methods summary for the |FLRW| class. If you are using
+IPython you can also use tab completion to print a list of the available
+methods. To do this, after importing the cosmology as in the above example,
```
Comment (mhvk) [inline docs/cosmology/index.rst:68]: I don't think one should be mixing in general introduction to (i)python here.  I'd delete "If you are ... tab key."

File: docs/cosmology/index.rst [hunk 8]
```diff
@@ -141,15 +147,13 @@ This can also be done more explicitly using units, which is recommended::
   >>> import astropy.units as u
   >>> cosmo = FlatLambdaCDM(H0=70 * u.km / u.s / u.Mpc, Tcmb0=2.725 * u.K, Om0=0.3)
 
-However, most of the parameters that accept units (``H0``, ``Tcmb0``)
-have default units, so unit quantities do not have to be used (with the
-exception of neutrino masses, where you must supply a unit if you want massive
-neutrinos).
+However, most of the parameters that accept units (``H0``, ``Tcmb0``) have
```
Comment (mhvk) [inline docs/cosmology/index.rst:150]: Maybe not mention this; with the units the code is clearer...

**Review by mhvk — APPROVED** [review]
Overall, this looks good, though there are a few places where I think the text could be cleaned up and made more concise. But perhaps that is better done in follow-up, since this PR seems to do mostly formatting. Since it is definitely better, I'll approve.

File: docs/cosmology/dev.rst (hunk 1 above)
Comment (nstarman) [reply docs/cosmology/dev.rst:10]: Agreed.

File: docs/cosmology/dev.rst (hunk 2 above)
Comment (nstarman) [reply docs/cosmology/dev.rst]: SGTM.

File: docs/cosmology/dev.rst (hunk 3 above)
Comment (nstarman) [reply docs/cosmology/dev.rst]: Yes. I've implemented your suggestion + a little more explanation.

File: docs/cosmology/dev.rst (hunk 4 above)
Comment (nstarman) [reply docs/cosmology/dev.rst:43]: Done.

File: docs/cosmology/dev.rst (hunk 5 above)
Comment (nstarman) [reply docs/cosmology/dev.rst]: Agreed.

File: docs/cosmology/index.rst (hunk 7 above)
Comment (nstarman) [reply docs/cosmology/index.rst]: Agreed.

File: docs/cosmology/index.rst (hunk 8 above)
Comment (nstarman) [reply docs/cosmology/index.rst]: Removed

**Comment by nstarman** [comment]
> Overall, this looks good, though there are a few places where I think the text could be cleaned up and made more concise. But perhaps that is better done in follow-up, since this PR seems to do mostly formatting.

Agreed! One of these days I'd like to redo the docs. Rewrite portions, restructure others. Move some stuff to separate pages. In particular I think the built-in realizations deserve their own sections. Particularly when we offer more than just the ML concordance fit.  (see project board for details)

> Since it is definitely better, I'll approve.

Thanks!

**Comment by pllim** [comment]
This was backported in  #12747 , which was originally the backport for  #12622 .

**Pull request merged**